app-modules/care-team/tests/CareTeamAutoSubscriptionTest.php

```diff
@@ -0,0 +1,84 @@
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\CareTeam\Tests;
+
+use App\Models\User;
+
+use function Tests\asSuperAdmin;
+
+use AdvisingApp\Prospect\Models\Prospect;
+
+use function Pest\Laravel\assertDatabaseHas;
+
+use Illuminate\Database\Eloquent\Collection;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\StudentDataModel\Models\Student;
+use AdvisingApp\StudentDataModel\Models\Contracts\Educatable;
+
+it('can auto subscribe user when adding in care team', function (array $careTeam, Collection $educatables) {
+    $superAdmin = User::factory()->licensed([Prospect::getLicenseType(), Student::getLicenseType()])->create();
+
+    asSuperAdmin($superAdmin);
+
+    $educatables->each(function (Educatable $educatable) use ($careTeam) {
+        $educatable->careTeam()->sync($careTeam);
+
+        $educatable->careTeam->each(function ($user) use ($educatable) {
+            assertDatabaseHas('care_teams', [
+                'user_id' => $user->getKey(),
+                'educatable_id' => $educatable->getKey(),
+                'educatable_type' => $educatable->getMorphClass(),
+            ]);
+
+            assertDatabaseHas('subscriptions', [
+                'user_id' => $user->getKey(),
+                'subscribable_id' => $educatable->getKey(),
+                'subscribable_type' => $educatable->getMorphClass(),
+            ]);
+        });
+    });
+})
+    ->with([
+        'for Prospect' => [
+            'careTeam' => fn () => User::factory()->licensed(LicenseType::cases())->count(1)->create()->pluck('id')->toArray(),
```

Comment:
Especially because this is the exact same for each data set I don't think it is needed to have this as a part of the dataset. It can just be in the test.
It is also a little confusing to call it careteam when it is a User.